plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/data/managers/AbstractTextPanelEditor.java

```diff
@@ -269,28 +310,65 @@ public <T> T getAdapter(Class<T> adapter) {
-            if (longContent) {
-                UIUtils.asyncExec(() -> {
-                    editor.setInput(new StringEditorInput("Empty", "", true, StandardCharsets.UTF_8.name()));
-                });
+            int maxContentSize = DBWorkbench.getPlatform().getPreferenceStore().getInt(ResultSetPreferences.RS_EDIT_MAX_TEXT_SIZE) * 1000;
+            if (editor == null) {
+                log.error("Editor is null or undefined");
+                return;
             }
-            UIUtils.asyncExec(() -> {
+            resetEditorInput();
+            if (value.getContentLength() > maxContentSize) {
+                showRestrictedContent(value);
+            } else {
+                showRegularContent(monitor);
+            }
+        } catch (Exception e) {
+            throw new DBException("Error loading text value", e);
+        } finally {
+            monitor.done();
+        }
+    }
 
-                if (textViewer != null && editor != null) {
+    private void resetEditorInput() {
+        // Load contents in two steps (empty + real in async mode). Workaround for some
+        // strange bug in StyledText in E4.13 (#6701)
+        UIUtils.asyncExec(() -> {
+            if (editor != null) {
+                messageBar.setText("");
+                messageBar.setImage(null);
+                editor.setInput(new StringEditorInput("Empty", "", true, StandardCharsets.UTF_8.name()));
+            }
+        });
+    }
+
+    private void showRegularContent(@NotNull DBRProgressMonitor monitor) throws DBException {
+        String encoding = getPanelSettings().get(PREF_TEXT_EDITOR_ENCODING);
+        if (encoding == null) {
+            encoding = StandardCharsets.UTF_8.name();
+        }
+        final ContentEditorInput textInput = new ContentEditorInput(valueController, null, null, encoding, monitor);
+        UIUtils.asyncExec(() -> {
+            if (editor != null) {
+                final TextViewer textViewer = editor.getTextViewer();
+                if (textViewer != null) {
                     StyledText textWidget = textViewer.getTextWidget();
-                    if (textWidget != null && longContent) {
+                    if (textWidget != null) {
                         GC gc = new GC(textWidget);
                         try {
-                            UIUtils.drawMessageOverControl(textWidget, gc, NLS.bind(ResultSetMessages.panel_editor_text_loading_placeholder_label, textInput.getContentLength()), 0);
+                            UIUtils.drawMessageOverControl(textWidget, gc,
+                                NLS.bind(ResultSetMessages.panel_editor_text_loading_placeholder_label, textInput.getContentLength()), 0);
                             editor.setInput(textInput);
+                            messageBar.setForeground(UIStyles.getDefaultTextForeground());
+                            messageBar.setText(String.format("Content size: %s Bytes", textInput.getContentLength()));
```

Comment:
I'm unsure whether this label should be always displayed or not. It's probably up to a discussion.

Reply:
Persistant panel looks empty and ugly without information, displaing content lenght is kind of interesting to have. But let's enchance it after UI/UX discussion.